docs/network-policy.md

```diff
@@ -0,0 +1,361 @@
+
+```
+_uuid               : 3f864884-cdb7-44be-a60e-e4f743afc9d0
+action              : allow
+direction           : to-lport
+external_ids        : {default-deny-policy-type=Ingress}
+log                 : false
+match               : "outport == @ingressDefaultDeny && arp"
+meter               : []
+name                : []
+priority            : 1001
+severity            : []
+
+```
+
+## **Applying the network policy to specific pods using `spec.podSelector`**
+
+In some cases only certain pods in a namespace may need to be selected by the NetworkPolicy. To handle this feature the `spec.podSelector` field can be used as follows 
+
+The `spec.podSelector` is a label selector, which can be either a list of labels (`app=nginx`) or a match expression. Only pods in the same namespace as the NetworkPolicy can be selected by it. The end result is a list of zero or more pods to which this NetworkPolicy's `ingress` and `egress` sections will be applied.
+
+For example, to block all traffic to and from a pod labeled with `app=demo` the following can be used 
+
+```
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: default-deny-all
+spec:
+  podSelector:
+    matchLabels:
+          app: demo
+  policyTypes:
+  - Ingress
+  - Egress
+```
+
+## **Applying Ingress and Egress Rules using `spec.ingress` and `spec.egress`** 
+
+In some cases we need to explicilty define what sources and sinks a pod is allowed to communicate with, to handle this feature the `spec.ingress` and `spec.egress` fields of a NeworkPolicy can be used 
+
+These sections contain a list of ingress or egress "peers" (the `from` section for `ingress` and the `to` section for `egress`) and a list of IP ports/protocols (the `ports` section) to or from which traffic should be allowed. Each list element of each section is logically OR-ed with other elements.
+
+Each `from`/`to` section can contain the following selectors 
+
+1. `namespaceSelector`: a label selector matching all pods in zero or more Namespaces
+2. `podSelector`: a label selector matching zero or more pods in the same Namespace as the NetworkPolicy
+3. `namespaceSelector` and `podSelector`: when both are present in an element, selects only pods matching the `podSelector` from namespaces matching the `namespaceSelector`
+4. `ipBlock`: an IP network in CIDR notation that can be either internal or external, with optional exceptions
+
+### `spec.ingress`
+
+Rules defined in `spec.ingress` can match on two main sections, 1.`spec.ingress.from` and 2.`spec.ingress.ports`
+
+1. `spec.ingress.from` 
+
+    Specifies FROM what sources a network policy will allow traffic 
+
+    It contains three selectors which are described further below 
+
+    - `spec.ingress.from.ipBlock` 
+        
+        The ipaddresses from which to allow traffic, contains fields `spec.ingress.from.ipBlock.cidr` to specify which ip address are allowed and 
```

Comment:
nit: be consistent with IP addresses. Here, it is `ipaddresses`, line 193 is `ip addresses`